Chatto/sources/ChatController/ChatMessages/ChatMessageCollectionAdapter.swift

```diff
@@ -0,0 +1,702 @@
+//
+// Copyright (c) Badoo Trading Limited, 2010-present. All rights reserved.
+//
+
+import UIKit
+
+public protocol ChatMessageCollectionAdapterProtocol: UICollectionViewDataSource, UICollectionViewDelegate {
+    var chatItemCompanionCollection: ChatItemCompanionCollection { get }
+    var delegate: ChatMessageCollectionAdapterDelegate? { get set }
```

Comment:
Do you think it would be possible to remove `delegate` properties here and in `ChatMessagesViewControllerProtocol`
Unless its needed to be updated after setting up objects graph, I think it would be benifitial to remove it from the protocol declaration.
We usually need to set it up once in the factory, when we know the type of the object, and could access this property without protocol.